### PR TITLE
fix(compiler): update css regex to support multiline comments

### LINKED ISSUE #14143: Css with comments parse incorrectly causing styles to be removed
**I'm submitting a ...**  (check one with "x")
```
[X] bug report => search github for a similar issue or PR before submitting
[ ] feature request
[ ] support request => Please do not submit support request here, instead see https://github.com/angular/angular/blob/master/CONTRIBUTING.md#question
```

**Current behavior**
My component has a templateUrl that points to an html file with a style tag on the top with comments, ex:
 /*align-items: center;
        box-sizing:border-box;
        max-width:100%;/*https://github.com/philipwalton/flexbugs */

I was using the version 2.4 without problems, when i started using 2.4.5 the component had no styles applied at all.

**Expected behavior**
Comments should be ignored and styles should be added to the component

**Minimal reproduction of the problem with instructions**

**What is the motivation / use case for changing the behavior?**
Creation of a component

**Please tell us about your environment:**
Windows 7, VS 2015, npm

* **Angular version:** 2.0.5

* **Browser:** [all | Chrome XX | Firefox XX | IE XX | Safari XX | Mobile Chrome XX | Android X.X Web Browser | iOS XX Safari | iOS XX UIWebView | iOS XX WKWebView ]

 
* **Language:** [all | TypeScript X.X | ES6/7 | ES5]

* **Node (for AoT issues):** `node --version` =

### PR DESCRIPTION
**Pull request by DzmitryShylovich**
Fixes #14143 and https://github.com/angular/angular/issues/15255

### CODE REVIEW

File: modules/@angular/compiler/src/style_url_resolver.ts [hunk 1]
```diff
@@ -43,5 +43,5 @@ export function extractStyleUrls(
 }
 
 const CSS_IMPORT_REGEXP = /@import\s+(?:url\()?\s*(?:(?:['"]([^'"]*))|([^;\)\s]*))[^;]*;?/g;
-const CSS_COMMENT_REGEXP = /\/\*.+?\*\//g;
+const CSS_COMMENT_REGEXP = /\/\*[^*]*\*+([^/*][^*]*\*+)*\//g;
```
Comment (gkalpak) [inline modules/@angular/compiler/src/style_url_resolver.ts]: You could make the group non-capturing, since you don't need it.

Comment (DzmitryShylovich) [inline modules/@angular/compiler/src/style_url_resolver.ts]: Good point, thx

**Review by mhevery — APPROVED** [review]

File: modules/@angular/compiler/src/style_url_resolver.ts [hunk 2]
```diff
@@ -43,5 +43,5 @@ export function extractStyleUrls(
 }
 
 const CSS_IMPORT_REGEXP = /@import\s+(?:url\()?\s*(?:(?:['"]([^'"]*))|([^;\)\s]*))[^;]*;?/g;
-const CSS_COMMENT_REGEXP = /\/\*.+?\*\//g;
```
Comment (vicb) [inline modules/@angular/compiler/src/style_url_resolver.ts]: use `[\s\S]+?` instead

**Comment by DzmitryShylovich** [comment]
@vicb done

**Comment by mhevery** [comment]
Closing since https://github.com/angular/angular/pull/18452 already added this functionality.

**Pull request closed without merging**